src/libraries/Common/src/System/Net/Http/Http2/Hpack/HPackDecoder.cs

```diff
@@ -466,13 +464,13 @@ int Decode(ref byte[] dst)
             _state = nextState;
         }
 
-        private HeaderField GetHeader(int index)
+        private ref HeaderField GetHeader(int index)
```

Comment:
If I'm not wrong this can introduce memory corruption in case we return dynamic value which is deleted later. What is your opinion?

Reply:
It won't introduce memory corruption, but you could end up pointing to stale or different header data.  So, not something that should be done unless there's a guarantee I'm not seeing that the referenced slot will remain containing the same data indefinitely.